### PR TITLE
Feature/refactor api around use of exceptions

### PR DESCRIPTION
**Pull request by mikonieminen**

### CODE REVIEW

File: lib/letters.ml [hunk 1]
```diff
@@ -85,59 +87,63 @@ let certs_of_pem_directory ?(ext = "crt") path =
...
+    in
+    let body =
+      match body with
+      | Plain text -> Mt.part ~header:plain_text_headers (stream_of_string text)
+      | Html text -> Mt.part ~header:html_headers (stream_of_string text)
+    in
+    Ok (Mt.make (Mrmime.Header.of_list headers) Mt.simple body)
+  with
+  | Invalid_email_address address -> Error (Printf.sprintf "Invalid email address: %s" address)
+  | _ -> Error "Unexpected error while trying to build an email message"
```
Comment (joseferben) [inline lib/letters.ml]: Use `Printexc.to_string` to stringify the exception, so we know what we went wrong.

Comment (mikonieminen) [reply lib/letters.ml]: Fixed and thanks, I was wondering how to do this.

Comment (joseferben) [reply lib/letters.ml]: I don't see the change in github

Comment (mikonieminen) [reply lib/letters.ml]: damn, I somehow managed to fail to push it. Fixed, but now Github has some issues and cannot push.

Comment (mikonieminen) [reply lib/letters.ml]: and now it worked, so fixed.

**Review by joseferben — APPROVED** [review]

**Pull request merged**